Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/TextInput.swift

```diff
@@ -120,15 +106,24 @@ private extension TextInput {
                         axis: .horizontal
                     )
                     .accessibilityIdentifier("\(element.label) Text Input")
+                    .focused($isFocused)
                     .keyboardType(keyboardType)
 #if os(visionOS)
                     // No need for hover effect since it will be applied
                     // properly at 'formInputStyle'.
                     .hoverEffectDisabled()
 #endif
+                    .onChange(isFocused) { isFocused in
+                        model.focusedElement = isFocused ? element : nil
+                    }
+                    .onChange(model.focusedElement) { focusedElement in
+                        // Another form input took focus
+                        if focusedElement != element {
+                            isFocused  = false
+                        }
+                    }
                 }
             }
-            .focused($isFocused)
```

Comment:
This appeared to be the issue's root; the modifier wasn't applying successfully onto `FullScreenTextInput`, so I've moved it down.